oauthproxy.go

```diff
@@ -686,6 +707,12 @@ func (p *OAuthProxy) isAllowedRoute(req *http.Request) bool {
 	return false
 }
 
+// isForwardedRequest is used to check if X-Forwarded-Host header exists or not
+func isForwardedRequest(req *http.Request, reverseProxy bool) bool {
```

Comment:
Is there a viable scenario where `X-Forwarded-Host` is not set, but `X-Forwarded-Proto` or `X-Forwarded-Uri` is set 🤔 

No idea -- just thinking out loud if that is a thing anywhere (beyond Traefik too).

I'm thinking for the future (in the scope creep PR we mentioned handling ReverseProxy better globally). If we embed the ReverseProxy state into the req context -- the `util.GetRequest*` method will handle this and the redirect builder could just be this snippet you have above potentially:

```
fmt.Sprintf("%s://%s%s", util.GetRequestProto(req), util.GetRequestHost(req), uri)
```